cob_omni_drive_controller/src/UndercarriageCtrlGeom.cpp

```diff
@@ -122,30 +102,31 @@ void CtrlData::setTarget(const PlatformState &plt_state){
     // Wheel has to move in direction of resulting velocity vector of steering axis
     double dAngGearSteerTarget1Rad = MathSup::atan4quad(dtempAxVelYRobMMS, dtempAxVelXRobMMS);
     // calculate corresponding angle in opposite direction (+180 degree)
-    double dAngGearSteerTarget2Rad = dAngGearSteerTarget1Rad + MathSup::PI;
-    MathSup::normalizePi(dAngGearSteerTarget2Rad);
+    double dAngGearSteerTarget2Rad = dAngGearSteerTarget1Rad + M_PI;
+    dAngGearSteerTarget2Rad = angles::normalize_angle(dAngGearSteerTarget2Rad);
 
     // calculate absolute value of rotational rate of driving wheels in rad/s
     double dVelGearDriveTarget1RadS = sqrt( (dtempAxVelXRobMMS * dtempAxVelXRobMMS) +
                                         (dtempAxVelYRobMMS * dtempAxVelYRobMMS) ) / geom_.dRadiusWheelMM;
     // now adapt to direction (forward/backward) of wheel
     double dVelGearDriveTarget2RadS = -dVelGearDriveTarget1RadS;
+ 
+    // calculate shortes distance for forward and backward facing wheel
+    double dist_current_to_desired = angles::shortest_angular_distance(state_.dAngGearSteerRad, dAngGearSteerTarget1Rad);
+    double dist_current_to_desired_inv = angles::shortest_angular_distance(state_.dAngGearSteerRad, dAngGearSteerTarget2Rad);
 
-
-    if(getWeightedDelta(state_.dAngGearSteerRad,  m_dAngGearSteerTargetRad, dAngGearSteerTarget1Rad)
-        <= getWeightedDelta(state_.dAngGearSteerRad, m_dAngGearSteerTargetRad, dAngGearSteerTarget2Rad))
+    //add shortes calculated distance to current position for new desired position
+    if(fabs(dist_current_to_desired) <= fabs(dist_current_to_desired_inv))
     {
-        // Target1 is "optimal"
         m_dVelGearDriveTargetRadS = dVelGearDriveTarget1RadS;
-        m_dAngGearSteerTargetRad = dAngGearSteerTarget1Rad;
+        m_dAngGearSteerTargetRad = state_.dAngGearSteerRad + dist_current_to_desired;
     }
     else
     {
-        // Target2 is "optimal"
         m_dVelGearDriveTargetRadS = dVelGearDriveTarget2RadS;
-        m_dAngGearSteerTargetRad = dAngGearSteerTarget2Rad;
+        m_dAngGearSteerTargetRad = state_.dAngGearSteerRad + dist_current_to_desired_inv;
     }
-
+    //TODO: test if this still works with velocity controller
```

Comment:
should be done before merging!

Reply:
tested with velocity controller on cob4-19. I think it is better than before. @ipa-fmw can you please take a look at cob4-19